packages/ui-lib/src/Indicators/Spinner.tsx

```diff
@@ -31,13 +30,13 @@ const rotate = keyframes`
   }
 `;
 
-const BaseCircle = styled.circle<{ styling: string }>`
-  stroke: ${({styling}) => `var(--spinner-${styling}, --grey-a8)` };
+const BaseCircle = styled.circle<{ styling: string, customColor?: string }>`
+  stroke: ${({styling, customColor}) => customColor ? customColor : `var(--spinner-${styling}-base, --grey-a8)` };
   fill: none;
 `;
 
-const RotatingCircle = styled.circle<{ r: number }>`
-  stroke: var(--white-1);
+const RotatingCircle = styled.circle<{ r: number, styling: string, customColor?: string }>`
+  stroke: ${({styling, customColor}) => customColor ? customColor : `var(--spinner-${styling}-top, --white-1)` };
   fill: none;
```

Comment:
I've fixed the var issue on fallback for non-custom colors. However with custom colors, if you provide an incorrect value it will just break.

That's something I don't think we can easily fix unless we force this to be only variable names or if we add another layer of function to check if it's a variable or not and process it accordingly, which I feel is getting heavy handed.

The reason why is that if you look at this...
`  stroke: ${({styling, customColor}) => customColor ? customColor : `var(--spinner-${styling}-top, var(--white-1))` };`

I'm basically taking the customColor and applying as the property. This means if it's incorrect, it breaks it. But you can add `#ff0000`, `red` or `var(--red-1)` all as options if you wanted to. But if we wrapped it in a `var` so you could handle fallback in the case of variables then we lose the ability to then allow standard values to be passed because they don't work in a var function.

As this is an advanced feature, it's on the engineer to check that their custom colors are applied correctly, I feel.

Reply:
Thank you for fixing the fallback color. Now it works well as fallback.

I tested the custom color with the examples provided but theme variables were not working for me.
Am I missing something?


**var(--red-1)**
<img width="689" alt="Screenshot 2024-12-02 at 13 43 40" src="https://github.com/user-attachments/assets/5fb2ac5e-11ff-4186-b928-190802c65e76">

**#ff0000**
<img width="680" alt="Screenshot 2024-12-02 at 13 43 27" src="https://github.com/user-attachments/assets/ca4d049e-a084-45dc-81f4-08f670b74f68">